WordPress/gutenberg
# Migrating block editor `BlockPatternsList` component

## What?

This PR updates [`BlockPatternsList`](https://github.com/WordPress/gutenberg/blob/3875618390ccf37c2df36d2ab4b7e249bc161660/packages/block-editor/src/components/block-patterns-list/index.js) in `@wordpress/block-editor` to use the updated `Composite` implementation from #54225.


## Why?

In #54225, an updated implementation of `Composite` was added to `@wordpress/components`. As per #55224, all consumers of `Composite` need to migrate from the old version to the new version.


## How?

 - Removes `__unstableComposite` imports from `@wordpress/components`
 - Adds private `Composite*` exports from `@wordpress/components`
 - Refactors `BlockPatternsList` and `BlockPattern` to use updated `Composite` components
 - Additionally renames list component to be consistent with codebase


## Testing Instructions

1. In the editor, open the block inserter sidebar
2. From the Patterns tab, open the patterns explorer via the "explore all patterns" button at the bottom
3. The patterns list should look no different

![The patterns explorer dialog, with "all patterns" selected in the sidebar](https://github.com/WordPress/gutenberg/assets/159848/858812b3-c99a-45e5-941d-3f606512c466)


### Testing Instructions for Keyboard

1. Inside the patterns explorer, dialog navigate to the "Block patterns" listbox
   - Note: this can be difficult using <kbd>tab</kbd>, as each sidebar option is a separate tab stop.
2. It should present as a singe tab stop, with each item navigable using arrow keys
